test/modules/geo-layers/tile-layer/tile-2d-header.spec.js

```diff
@@ -46,3 +46,21 @@ test('Tile2DHeader#Cancel request if not selected', async t => {
   t.ok(tile2.isCancelled, 'Second request was cancelled');
   t.end();
 });
+
+test('Tile2DHeader#Abort quickly', async t => {
+  const requestScheduler = new RequestScheduler({throttleRequests: true, maxRequests: 1});
+
+  const getTileData = () => null;
+  const onTileLoad = () => null;
+  const onTileError = () => null;
+
+  const tile = new Tile2DHeader({onTileLoad, onTileError});
+
+  // Await later so that the abort could go off before the getTileData call.
+  const loader = tile._loadData(getTileData, requestScheduler);
+  tile.abort();
+  await loader;
```

Comment:
Add `await new Promise((resolve) => setTimeout(resolve, 0));`, requestScheduler resolves asynchronously after the loader

Reply:
https://github.com/visgl/deck.gl/pull/5044/files#diff-5dd332fb839a9666cc7d7489bfa32eda9f101a0f17b44034ea90a5d132d61b74R53

Reply:
That's not the same thing. After loader is resolved, requestScheduler's `_issueRequest` is still pending.

Reply:
Ok, I can add this - I don't fully understand why it's needed.  Can you explain a bit?

Reply:
I suggest you run this test with and without your change to `tile-2d-header.js` - If the test is constructed correctly, it should fail before the change and succeed after.

Reply:
Sorry ignore my previous suggestion - I thought you were calling `loadData`

Reply:
Ok, yes, I was just checking this and was confused.  No worries. I will revert the loader change.